bigquery_etl/cli/query.py

```diff
@@ -81,6 +81,7 @@
 DEFAULT_CHECKS_FILE_NAME = "checks.sql"
 VIEW_FILE = "view.sql"
 MATERIALIZED_VIEW = "materialized_view.sql"
+NBR_DAYS_RETAINED = 775
```

Comment:
I guess this is a good start. Though I wonder if we can use the `expiration_days` metadata to determine how far back the backfill should go. The retention can be different for different datasets, like here: https://github.com/mozilla/bigquery-etl/blob/7cdeef94d497dbcde97bccf0078f4ab0cba9da00/sql/moz-fx-data-shared-prod/braze_external/changed_products_sync_v1/metadata.yaml#L18

Ideally, the smallest retention range based on all upstream tables should be used here.

Reply:
I agree this would be a really good iteration idea - would be much more accurate and prevent more issues.